lib/caching-precompiler.js

```diff
@@ -30,7 +30,11 @@ class CachingPrecompiler {
 		this.transform = this._createTransform();
 	}
 
-	precompileFile(filePath) {
+	precompileFile(filePath, excludeExtensions) {
+		if (excludeExtensions.includes(path.extname(filePath))) {
```

Comment:
I think this should be guarded against in the calling code.

Reply:
Good idea moving this to be checked here https://github.com/avajs/ava/blob/master/api.js#L134